src/init.cpp

```diff
@@ -185,8 +186,13 @@ void updateConfigurationValue(const pika::program_options::variables_map& vm, T&
   }
 
   const std::string dlaf_cmdline_option = "dlaf:" + cmdline_option;
-  if (vm.count(dlaf_cmdline_option)) {
-    var = parseFromCommandLine<T>::call(vm, dlaf_cmdline_option);
+  if constexpr (std::is_same_v<T, bool>) {
+    var = var || vm.count(dlaf_cmdline_option) > 0;
+  }
+  else {
+    if (vm.count(dlaf_cmdline_option)) {
+      var = parseFromCommandLine<T>::call(vm, dlaf_cmdline_option);
+    }
```

Comment:
This is a bit funny but a result of the `--dlaf:print-config` option being a flag, not an option that takes `true/false`. This also assumes now that if we have binary configuration options, the corresponding command line option is simply a flag that toggles the option to true.

Do you think this is ok or would you prefer changing the command line option to be similar to the environment variable (which accepts `true/1/on/false/...`)?

Reply:
Do we have many things that are ON by default and potentially need to be turned OFF? Those are the ugly cases with the flag approach, which I think it's otherwise great if there is a majority of OFF->ON.

Reply:
I think the only other flag we have is `--dlaf:no-mpi-pool` which is binary, and negated for good measure. However, pika's MPI polling will replace this hopefully soon so I wouldn't consider this flag for this discussion.

Reply:
Agreed in meeting that this is ok behaviour.